### ISSUE TITLE
LigerFusedLinearCrossEntropyFunction uses wrong None check

### ISSUE THREAD

**Issue by brandon-oai** [linkedin/Liger-Kernel#1156]
### 🐛 Describe the bug

The forward in `LigerFusedLinearCrossEntropyFunction` does the following

```
        ctx.save_for_backward(
            grad_input.detach(),
            grad_weight.detach() if grad_weight is not None else None,
            grad_bias.detach() if bias is not None else None,
        )
```

When running this function with gradients off but w/ bias not None, `grad_bias` is None, so this throws an error.

### Versions

On `liger-kernel` v0.7.0, this should generalize across hardware
